### DISCUSSION TITLE
Multi-threading

### DISCUSSION THREAD

**Discussion by KeenanFernandes2000** [jakobhoeg/nextjs-ollama-llm-ui#62] [Q&A] [answered]
Is there a way to configure the code to cater to multiple user at the same time, I have server that can support the load however when multiple people try to access the endpoint it goes in a sequential mode and not multi-threaded.

**Comment by jakobhoeg** [comment, marked as answer]
Hey, @KeenanFernandes2000 !

Yes, it is possible in v.0.1.33 of Ollama. You need to set some environmental variables. Check out [this](https://github.com/ollama/ollama/releases/tag/v0.1.33) link, specifically under `Experimental concurrency features`.

`OLLAMA_NUM_PARALLEL`: Handle multiple requests simultaneously for a single model
`OLLAMA_MAX_LOADED_MODELS`: Load multiple models simultaneously

Example:

`OLLAMA_NUM_PARALLEL=4 OLLAMA_MAX_LOADED_MODELS=4`

**Reply by KeenanFernandes2000** [reply]
@jakobhoeg thank you for the quick response and with the source. Appreciate it 🙏
